### ISSUE TITLE
Notebooks synchronization issues

### ISSUE THREAD

**Issue by akhenry** [nasa/openmct#4737]
#### Summary
Two issues have been observed in testing and demos that relate to notebook object synchronization, aka "multi-user notebooks" that are regressions since 1.7.8.

1. Users will sometimes be unable to create new entries. Instead a new row will be created and focused, but will then immediately disappear.
2. Notebooks will sometimes stop showing remote changes made by other users altogether.

#### Expected vs Current Behavior
1. Notebooks should always allow new entries to be created
2. Notebooks should always reflect changes made my other users asynchronously

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [X] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [X] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Steps to Reproduce
For issue 1. see https://github.com/nasa/openmct/issues/3881#issuecomment-1007002164
For issue 2 see below:
1. Open the same notebook in two tabs or windows
2. Add a new entry to one notebook (if you are not blocked by the first issue)
3. Add a new entry to the other notebook. Observe that this new entry does not appear in the first notebook.

#### Environment
* Open MCT Version: 324318793ddeb46e4f0a5c41d7e8b7d2b61309d0
* Deployment Type: Dev
* OS: MacOS
* Browser: Chrome 97.0.4692.71 

## Testing instructions

Verify the issues are no longer observed by following the reproduction steps above.

**Comment by charlesh88** [comment]
I've observed that Notebooks will sync their views when the Notebook is in the main view in both instances, but won't when the Notebook is in a Display or Flexible Layout.

**Comment by akhenry** [comment]
@charlesh88 Good info, thanks. I believe [I have a fix for most of the notebook sync issues](https://github.com/nasa/openmct/pull/4738), I will also test the scenario you describe.

**Comment by akhenry** [comment]
Testing instructions:

Verify the issues are no longer observed by following the reproduction steps above.

**Comment by charlesh88** [comment]
Testathon 1-24-22: verified partially fixed. 
- [x] Main view > main view
- [x] Notebook in Display Layout > main view
- [x] Main view > Notebook in Display Layout 
- [x] Notebook in Flex Layout > main view
- [ ] Main view > Notebook in Flex Layout 

When a Notebook is in a Flex Layout, updates to it are synced out to a main view. But updates in the main view are not synced back in to the Flex instance, and if updates are added to the flex Notebook, entries in the main view will be wiped out.

<img width="2034" alt="Screen Shot 2022-01-24 at 2 50 26 PM" src="https://user-images.githubusercontent.com/1056412/150878684-79ead2ab-b50a-4983-bc3f-a9e5d648b1aa.png">

**Comment by akhenry** [comment]
@charlesh88 Oh, this is a good catch. Turns out that the problem is that, unlike display layouts, flexible layouts are not tracking _any_ external changes in _any_ sub objects. The problem is not contained to notebooks.

Here's another way to reproduce the same issue, but without notebooks:
1. Open a flexible layout
2. Expand the flexible layout in the tree
3. Right click on an object in the layout from the tree and use the properties dialog to rename it
4. Observe that the name updates in the tree, but not in the flexible layout.
5. Now renamed an object in the flexible layout
6. Note that the name changes in the flexible layout _and_ in the tree.

Repeat the steps above in a display layout and observe that renaming objects works in both directions.

I'm going to create a separate issue to track this broader issue.
